### ISSUE TITLE
`prefer-math-min-max` issue with luxon objects

### ISSUE THREAD

**Issue by eldair** [sindresorhus/eslint-plugin-unicorn#2562]
Prefer math min max should not be reported for luxon objects because Math.min/max returns timestamp instead of original object which causes errors if not corrected and also there is loss of other information if object is recreated from Math result (like timezone)

prefer-math-min-max

```js
let start = DateTime.local().startOf('day').minus({days: 5});
let end = DateTime.local().startOf('day').minus({days: 1});

let min = Math.min(start, end);
// min is now a timestamp and not an object.
```

**Comment by axetroy** [comment]
Since there is no Type-aware in the Plugin, we cannot know its type

This seems normal to us, using implicit `valueOf()` is not the recommended way

```js
let start = DateTime.local().startOf('day').minus({days: 5});
let end = DateTime.local().startOf('day').minus({days: 1});

// ❌ implicit `valueOf()`
let min = start > end ? end : start;
```

You should write to this

```js
let start = DateTime.local().startOf('day').minus({days: 5});
let end = DateTime.local().startOf('day').minus({days: 1});

// ✅
let min = start.valueOf() > end.valueOf() ? end : start;
```

**Comment by eldair** [comment]
@axetroy thanks for getting back to me.
I actually like your suggestion but that line will not report I have used implicit valueOf, only math errors.

Maybe autofix should be turned off for anything except numbers to Math.min/max, the way it works seems unsafe.

**Comment by eldair** [comment]
If disabling auto fix for non numbers is something you would consider I could make a pull request

**Comment by james-hu** [comment]
> If disabling auto fix for non numbers is something you would consider I could make a pull request

Yes, I think it should be disabled for non-number variable. I've seen it triggered on string variables and auto-fixed to use Math.min(string1, string2)

**Comment by sindresorhus** [comment]
Going to pass on this as it requires types to improve, and I don't see strong enough reasons to add options or change behavior.

**Comment by eldair** [comment]
Hi @sindresorhus would this really become worse if disabled for non number variables?

**Comment by sindresorhus** [comment]
Yes. It would make it worse for 90% of users, just because of an edge-case. Everything is a balance. I'm not going to change this. You can either disable the rule globally or inline.

**Comment by eldair** [comment]
> Yes. It would make it worse for 90% of users, just because of an edge-case. Everything is a balance. I'm not going to change this. You can either disable the rule globally or inline.

Okay, I didn't mean that as an attack, just curious.. My main grip with this was that it can break code if the change slips by.

**Comment by clshortfuse** [comment]
Just hit this similarly and wondering if there were any solution considering it reminds me of https://github.com/sindresorhus/eslint-plugin-unicorn/pull/3213

found this ugly string comparison pattern in code and autofix broke it. 
```js
// Repro for unsafe `unicorn/prefer-math-min-max` autofix:
//
// ISO date strings in YYYY-MM-DD format compare correctly lexicographically.
const startDate = '2026-06-15';
const weekStart = '2026-06-01';

const clamped = startDate < weekStart ? weekStart : startDate;
// => '2026-06-15'

// `unicorn/prefer-math-min-max --fix` rewrites that shape to:
const autofixed = Math.max(startDate, weekStart);
// => NaN, because Math.max coerces date strings to numbers
```

Not sure if we have type support yet, but I have to disable the rule because of autofix.

Edit: Of course, now i'm looking for a eslint to prevent `<` comparisons on strings.
